src/neml2/models/ScalarTrialStressUpdate.cxx

```diff
@@ -0,0 +1,55 @@
+#include "neml2/models/ScalarTrialStressUpdate.h"
+
+namespace neml2
+{
+register_NEML2_object(ScalarTrialStressUpdate);
+
+OptionSet
+ScalarTrialStressUpdate::expected_options()
+{
+  OptionSet options = Model::expected_options();
+
+  options.set_input("elastic_trial_stress") = VariableName("forces", "s0");
+  options.set("elastic_trial_stress").doc() = "Initial elastic trial stress";
+
+  options.set_input("inelastic_strain") = VariableName("state", "ep");
+  options.set("inelastic_strain").doc() = "Current guess for the inelastic strain rate";
+
+  options.set_output("updated_trial_stress") = VariableName("state", "s");
+  options.set("updated_trial_stress").doc() = "Trial stress corrected for inelastic deformation";
+
+  options.set_parameter<CrossRef<Scalar>>("youngs_modulus");
+  options.set("youngs_modulus").doc() = "Young's modulus";
+
+  options.set_parameter<CrossRef<Scalar>>("poisson_ratio");
+  options.set("poisson_ratio").doc() = "Poisson's ratio";
+
+  options.set<bool>("_use_AD_first_derivative") = true;
+  options.set<bool>("_use_AD_second_derivative") = true;
+  return options;
+}
+
+ScalarTrialStressUpdate::ScalarTrialStressUpdate(const OptionSet & options)
+  : Model(options),
+    _elastic_trial_stress(declare_input_variable<Scalar>("elastic_trial_stress")),
+    _inelastic_strain(declare_input_variable<Scalar>("inelastic_strain")),
+    _inelastic_strain_old(declare_input_variable<Scalar>(_inelastic_strain.name().old())),
+    _updated_trial_stress(declare_output_variable<Scalar>("updated_trial_stress")),
+    _E(declare_parameter<Scalar>("E", "youngs_modulus", /*allow_nonlinear=*/true)),
+    _nu(declare_parameter<Scalar>("nu", "poisson_ratio", /*allow_nonlinear=*/true))
+{
+}
+
+void
+ScalarTrialStressUpdate::set_value(bool out, bool /*dout_din*/, bool /*d2out_din2*/)
```

Comment:
Don't we need the chain rule derivatives here?

Reply:
Yeah, we do. I though I had added use ad options here...

Reply:
And I did. What are you saying is missing then?

Reply:
Before we merge it would be nice to have the analytic derivatives included, like all the other models.

Reply:
I implemented the derivatives